src/sds.c

```diff
@@ -748,18 +748,15 @@ void sdsrange(sds s, ssize_t start, ssize_t end) {
         end = len+end;
         if (end < 0) end = 0;
     }
-    newlen = (start > end) ? 0 : (end-start)+1;
-    if (newlen != 0) {
-        if (start >= (ssize_t)len) {
-            newlen = 0;
-        } else if (end >= (ssize_t)len) {
-            end = len-1;
-            newlen = (start > end) ? 0 : (end-start)+1;
-        }
+    if (end >= (ssize_t)len) {
+        end = len - 1;
+    }
+    if (start >= (ssize_t)len) {
+        newlen = 0;
     } else {
-        start = 0;
+        newlen = (start > end) ? 0 : (end-start)+1;
     }
-    if (start && newlen) memmove(s, s+start, newlen);
+    if (newlen) memmove(s, s+start, newlen);
```

Comment:
Oops! Maybe we should not leave the check for start(start &&) in the if condition